### ISSUE TITLE
Using the -Dquarkus.kubernetes.namespace option does not update subjects

### ISSUE THREAD

**Issue by shawkins** [quarkusio/quarkus#16763]
## Describe the bug
When the src/main/kubernetes.yml contains a ClusterRoleBinding referencing a ServiceAccount in the same file and you try to override the namespace via the quarkus.kubernetes.namespace property, the referenced ServiceAccount namespace will not be updated.

## To Reproduce

Use a src/main/kubernetes.yml like:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: sa-name
  labels:
    app: sa-name
...
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: cr-name
subjects:
  - kind: ServiceAccount
    name: sa-name
    namespace: default-namespace
...
```
A maven package with -Dquarkus.kubernetes.namespace=expected

Will generate a kubernetes.yml with:
```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
...
subjects:
- kind: ServiceAccount
  name: sa-name
  namespace: default-namespace
```
Rather than updating the subject to the expected namespace - all of the top level resources though will specify the expected namespace.

### Quarkus version or git rev
1.11.1.Final

**Comment by geoand** [comment]
Have you tried with the latest 1.11 version of Quarkus?

**Comment by geoand** [comment]
@iocanel this sounds like one that needs to be fixed in the 1.11 branch.

Can you take care of it please?

**Comment by shawkins** [comment]
> Have you tried with the latest 1.11 version of Quarkus?

I have not yet.
